src/components/study-container.jsx

```diff
@@ -438,49 +438,28 @@ export function StudyContainer({ view, onChangeTab }) {
         return ws;
     }, []);
 
-    useEffect(() => {
-        // create ws at mount event
-        wsRef.current = connectNotificationsWsUpdateDirectories();
-
-        wsRef.current.onmessage = function (event) {
-            const eventData = JSON.parse(event.data);
-            dispatch(studyUpdated(eventData));
-            if (eventData.headers) {
-                if (
-                    eventData.headers['notificationType'] ===
-                    directoriesNotificationType.UPDATE_DIRECTORY
-                ) {
-                    // TODO: this receives notifications for all the public directories and all the user's private directories
-                    // At least we don't fetch everytime a notification is received, but we should instead limit the
-                    // number of notifications (they are sent to all the clients every time). Here we are only
-                    // interested in changes in parent directories of the study (study is moved, or any parent is moved
-                    // or renamed)
-                    if (
-                        studyParentDirectoriesUuidsRef.current.includes(
-                            eventData.headers['directoryUuid']
-                        )
-                    ) {
-                        fetchStudyPath();
-                    }
-                }
+    useListener(WS_URL_KEYS.DIRECTORIES, (event) => {
+        const eventData = JSON.parse(event.data);
+        dispatch(studyUpdated(eventData));
```

Comment:
I think the objective was to get rid of studyUpdateForce in the reducer ? maybe we got lost in our deliberations, should we refocus on this ?

Reply:
The main goal of this draft PR is to discuss the usage of the websocket component.

The rework from the current usage to this should be done in another PR after merging.